src/status_im/hardwallet/core.cljs

```diff
@@ -49,17 +93,151 @@
   [{:keys [db]} enabled?]
   {:db (assoc-in db [:hardwallet :nfc-enabled?] enabled?)})
 
-(fx/defn navigate-to-connect-screen [{:keys [db] :as cofx}]
+(fx/defn status-hardwallet-option-pressed [{:keys [db] :as cofx}]
   (fx/merge cofx
             {:hardwallet/check-nfc-enabled    nil
              :hardwallet/register-card-events nil
-             :db                              (assoc-in db [:hardwallet :setup-step] :begin)}
+             :db                              (-> db
+                                                  (assoc-in [:hardwallet :setup-step] :begin)
```

Comment:
settings-screen-did-load does the same. Refactoring?

Reply:
@jeluard not exactly the same. `:setup-step` being set to different value.